pkg/engine/context/context.go

```diff
@@ -129,8 +128,8 @@ func (ctx *context) AddRequest(request *admissionv1.AdmissionRequest) error {
 	return addToContext(ctx, request, "request")
 }
 
-func (ctx *context) AddVariable(key, value string) error {
-	return ctx.addJSON(pkgcommon.VariableToJSON(key, value))
+func (ctx *context) AddVariable(key string, value interface{}) error {
+	return addToContext(ctx, value, strings.Split(key, ".")...)
```

Comment:
Isn’t this going to be a problem if the key contains dots ?

Reply:
True, that will cause issues if the top level key contains dots.

I will create an issue to track this.